File: pdu/python/drone_srv_msgs/pdu_cdr_conv_DroneLandResponse.py
from .pdu_pytype_DroneLandResponse import DroneLandResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DroneLandResponse(writer: CdrWriter, src: DroneLandResponse):
    writer.write_bool(src.ok)
    writer.write_string(src.message)


def cdr_body_to_py_DroneLandResponse(reader: CdrReader, dst: DroneLandResponse):
    dst.ok = reader.read_bool()
    dst.message = reader.read_string()
    return dst


def py_to_cdr_DroneLandResponse(src: DroneLandResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DroneLandResponse(writer, src)
    return writer.bytes()


def cdr_to_py_DroneLandResponse(cdr_payload) -> DroneLandResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DroneLandResponse()
    return cdr_body_to_py_DroneLandResponse(reader, dst)

File: pdu/javascript/drone_srv_msgs/pdu_cdr_conv_DroneLandResponse.js
import { DroneLandResponse } from './pdu_jstype_DroneLandResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDroneLandResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DroneLandResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_bool(src.ok);
        writer.write_string(src.message);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DroneLandResponse } dst
     * @returns { DroneLandResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.ok = reader.read_bool();
        dst.message = reader.read_string();
        return dst;
    }

    /**
     * @param { DroneLandResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DroneLandResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DroneLandResponse());
    }
}
